### PR TITLE
Basic fault injector skeleton

### LINKED ISSUE #53: Basic fault injector skeleton
## User stories
As a developer of the fault injector I want a skeleton of the methods and classes of the fault injector in order to implement the basic fault injector.

## Dependencies
Depends on: 
 - #18 

## Acceptance Criteria
- [x] Add class fault injector
- [x] Add possibility to add fault injector configs
- [x] add possibility to store components in which fault should be injected
- [x] adapt method next_tick
~~- [x] setup different types of fault classes~~
    ~~faults affecting infrastructure stored in yaramo~~
    ~~- [x] faults affecting simulation objects like trains~~

### PR DESCRIPTION
**Pull request by instant-sky**
Fixes #53 

Description (what might a PO or reviewer want to know?)

## PR checklist

- [x] Acceptance criteria fulfilled
- [ ] Additional features are tested
- [x] Docs (code, wiki & diagrams) updated
- [ ] Breaking changes anounced
- [x] Dev-branch has been merged into local branch to resolve conflicts
- [x] Tests and linter have passed AFTER local merge
- [ ] Another dev reviewed and approved

### CODE REVIEW

File: src/fault_injector/fault_injector.py
Comment (CR1337) [inline src/fault_injector/fault_injector.py:1]: Why are you importing `Type`?

File: src/fault_injector/fault_types/fault.py
Comment (CR1337) [inline src/fault_injector/fault_types/fault.py:1]: Shouldn't the attributes of Fault have type hints? Also, are they really all private?

File: src/fault_injector/fault_types/train_fault.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+from src.component import Component
+from src.fault_injector.fault_types.fault import Fault
+
+
+class TrainFault(Fault):
+    """A fault affecting trains."""
+
+    @classmethod
+    def from_json(cls, json_object: str) -> "Fault":
```
Comment (SaturnHafen) [inline src/fault_injector/fault_types/train_fault.py]:
```suggestion
    def from_json(cls, json_object: str) -> Fault:
```
There should be no need to specifiy the return type as a string

File: src/fault_injector/fault_types/fault.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    start_tick: int = None
+    end_tick: int = None
+    component: Component = None
+    affected_element_ID: int = None
+    description: str = "injected fault"
+
+    @classmethod
+    @abstractmethod
+    def from_json(cls, json_object: str) -> "Fault":
```
Comment (SaturnHafen) [inline src/fault_injector/fault_types/fault.py:17]:
```suggestion
    def from_json(cls, json_object: str) -> Fault:
```

same as below

File: src/fault_injector/fault_types/train_fault.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        """Constructs a TrainFault from a JSON object
+
+        :param json_object: The JSON object
+        :type json_object: str
```
Comment (SaturnHafen) [inline src/fault_injector/fault_types/train_fault.py:13]: Do we need the type annotations in the docs?

**Review by SaturnHafen — COMMENTED** [review]
Looks good so far!

It does not look like we are able to implement tests here :thinking:

File: src/fault_injector/fault_types/train_fault.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        :return: a TrainFault
+        :rtype: Fault
+        """
+        raise NotImplementedError()
+
+    def inject_fault(component: Component):
+        """inject TrainFault into the given component
+
+        :param component: _description_
```
Comment (SaturnHafen) [inline src/fault_injector/fault_types/train_fault.py]: Either remove the `param` line or add documentation

**Review by SaturnHafen — APPROVED** [review]
Looks good :+1:

**Pull request merged**